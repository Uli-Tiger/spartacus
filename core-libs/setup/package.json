{
  "name": "@spartacus/setup",
  "version": "6.4.0-1",
  "description": "Includes features that makes Spartacus and it's setup easier and streamlined.",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "setup",
    "recipe"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/core-libs/setup",
  "license": "Apache-2.0",
  "scripts": {
    "test": "../../node_modules/.bin/jest --config ./jest.config.js"
  },
  "dependencies": {
    "tslib": "^2.4.0"
  },
  "peerDependencies": {
    "@angular/core": "^15.2.4",
    "@spartacus/cart": "6.4.0-1",
    "@spartacus/core": "6.4.0-1",
    "@spartacus/order": "6.4.0-1",
    "@spartacus/user": "6.4.0-1"
  },
  "optionalDependencies": {
    "@angular/platform-server": "^15.2.4",
    "@nguniversal/express-engine": "^15.2.0",
    "express": "^4.18.2"
  },
  "publishConfig": {
    "access": "public"
  }
}
