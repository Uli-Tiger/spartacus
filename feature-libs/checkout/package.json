{
  "name": "@spartacus/checkout",
  "version": "6.8.0",
  "description": "Checkout feature library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "checkout"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/feature-libs/checkout",
  "license": "Apache-2.0",
  "exports": {
    ".": {
      "sass": "./_index.scss"
    }
  },
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^15.2.9",
    "@angular/common": "^15.2.9",
    "@angular/core": "^15.2.9",
    "@angular/forms": "^15.2.9",
    "@angular/router": "^15.2.9",
    "@ng-select/ng-select": "^10.0.3",
    "@ngrx/store": "^15.3.0",
    "@spartacus/cart": "6.8.0",
    "@spartacus/core": "6.8.0",
    "@spartacus/order": "6.8.0",
    "@spartacus/schematics": "6.8.0",
    "@spartacus/storefront": "6.8.0",
    "@spartacus/styles": "6.8.0",
    "@spartacus/user": "6.8.0",
    "bootstrap": "^4.6.2",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
