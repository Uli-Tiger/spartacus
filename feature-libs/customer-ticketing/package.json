{
  "name": "@spartacus/customer-ticketing",
  "version": "6.5.0-1",
  "description": "Customer-Ticketing library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "customer-ticketing"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/feature-libs/customer-ticketing",
  "license": "Apache-2.0",
  "exports": {
    ".": {
      "sass": "./_index.scss"
    }
  },
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^15.2.9",
    "@angular/common": "^15.2.9",
    "@angular/core": "^15.2.9",
    "@angular/forms": "^15.2.9",
    "@angular/router": "^15.2.9",
    "@spartacus/cart": "6.5.0-1",
    "@spartacus/core": "6.5.0-1",
    "@spartacus/schematics": "6.5.0-1",
    "@spartacus/storefront": "6.5.0-1",
    "@spartacus/styles": "6.5.0-1",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
