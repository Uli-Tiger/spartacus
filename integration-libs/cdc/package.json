{
  "name": "@spartacus/cdc",
  "version": "6.8.0",
  "description": "Customer Data Cloud Integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "customer data cloud",
    "gigya",
    "cdc"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/cdc",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^15.2.9",
    "@angular/common": "^15.2.9",
    "@angular/core": "^15.2.9",
    "@angular/forms": "^15.2.9",
    "@angular/router": "^15.2.9",
    "@ng-select/ng-select": "^10.0.3",
    "@ngrx/effects": "^15.3.0",
    "@ngrx/store": "^15.3.0",
    "@spartacus/asm": "6.8.0",
    "@spartacus/core": "6.8.0",
    "@spartacus/organization": "6.8.0",
    "@spartacus/schematics": "6.8.0",
    "@spartacus/storefront": "6.8.0",
    "@spartacus/user": "6.8.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
