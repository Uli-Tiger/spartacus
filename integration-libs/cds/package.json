{
  "name": "@spartacus/cds",
  "version": "6.6.0",
  "description": "Context Driven Service library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "merchandising",
    "segmentation",
    "context-driven services",
    "cds"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/cds",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"src/schematics/**/*.js\" \"src/schematics/**/*.js.map\" \"src/schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^15.2.9",
    "@angular/common": "^15.2.9",
    "@angular/core": "^15.2.9",
    "@angular/router": "^15.2.9",
    "@ngrx/store": "^15.3.0",
    "@spartacus/cart": "6.6.0",
    "@spartacus/core": "6.6.0",
    "@spartacus/order": "6.6.0",
    "@spartacus/schematics": "6.6.0",
    "@spartacus/storefront": "6.6.0",
    "@spartacus/tracking": "6.6.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./src/schematics/collection.json"
}
