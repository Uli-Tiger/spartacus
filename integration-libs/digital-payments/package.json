{
  "name": "@spartacus/digital-payments",
  "version": "6.4.0",
  "description": "Digital Payments Integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "digital payments",
    "payments"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/digital-payments",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.4.0"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^15.2.4",
    "@angular/common": "^15.2.4",
    "@angular/core": "^15.2.4",
    "@angular/forms": "^15.2.4",
    "@angular/router": "^15.2.4",
    "@ng-select/ng-select": "^10.0.3",
    "@spartacus/cart": "6.4.0",
    "@spartacus/checkout": "6.4.0",
    "@spartacus/core": "6.4.0",
    "@spartacus/schematics": "6.4.0",
    "@spartacus/storefront": "6.4.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
