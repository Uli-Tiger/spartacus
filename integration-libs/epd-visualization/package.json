{
  "name": "@spartacus/epd-visualization",
  "version": "6.7.0",
  "description": "SAP Enterprise Product Development Visualization integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "epd-visualization",
    "2d",
    "3d",
    "visual",
    "spare",
    "part",
    "picking",
    "epd"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/epd-visualization",
  "license": "Apache-2.0",
  "exports": {
    ".": {
      "sass": "./_index.scss"
    }
  },
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^15.2.9",
    "@angular/common": "^15.2.9",
    "@angular/core": "^15.2.9",
    "@angular/forms": "^15.2.9",
    "@angular/router": "^15.2.9",
    "@sapui5/ts-types-esm": "1.120.1",
    "@spartacus/cart": "6.7.0",
    "@spartacus/core": "6.7.0",
    "@spartacus/schematics": "6.7.0",
    "@spartacus/storefront": "6.7.0",
    "@spartacus/styles": "6.7.0",
    "bootstrap": "^4.6.2",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
