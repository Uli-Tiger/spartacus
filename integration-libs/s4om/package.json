{
  "name": "@spartacus/s4om",
  "version": "6.3.0-1",
  "description": "S/4HANA Order Management (b2b feature)",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "s4om"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.4.0"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^15.2.4",
    "@angular/common": "^15.2.4",
    "@angular/core": "^15.2.4",
    "@spartacus/cart": "6.3.0-1",
    "@spartacus/core": "6.3.0-1",
    "@spartacus/requested-delivery-date": "6.3.0-1",
    "@spartacus/schematics": "6.3.0-1",
    "@spartacus/storefront": "6.3.0-1",
    "@spartacus/styles": "6.3.0-1",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
