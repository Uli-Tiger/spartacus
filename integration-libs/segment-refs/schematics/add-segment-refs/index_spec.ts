/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  generateDefaultWorkspace,
  LibraryOptions as SegmentRefsOptions,
  SEGMENT_REFS_FEATURE_NAME,
  segmentRefsFeatureModulePath,
  SPARTACUS_SCHEMATICS,
  SPARTACUS_SEGMENT_REFS,
  TRACKING_PERSONALIZATION_FEATURE_NAME,
  trackingPersonalizationFeatureModulePath,
} from '@spartacus/schematics';
import * as path from 'path';
import { peerDependencies } from '../../package.json';

const collectionPath = path.join(__dirname, '../collection.json');

describe('Spartacus segment-refs schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner(
    SPARTACUS_SEGMENT_REFS,
    collectionPath
  );

  let appTree: UnitTestTree;

  const libraryNoFeaturesOptions: SegmentRefsOptions = {
    project: 'schematics-test',
    lazy: false,
    features: [],
  };

  const segmentRefsFeatureOptions: SegmentRefsOptions = {
    ...libraryNoFeaturesOptions,
    features: [SEGMENT_REFS_FEATURE_NAME],
  };

  describe('Without features', () => {
    beforeAll(async () => {
      appTree = await generateDefaultWorkspace(schematicRunner, appTree);
      appTree = await schematicRunner.runSchematic(
        'ng-add',
        libraryNoFeaturesOptions,
        appTree
      );
    });

    it('should not create any of the feature modules', () => {
      expect(appTree.exists(segmentRefsFeatureModulePath)).toBeFalsy();
    });
  });

  describe('Segment-refs feature', () => {
    describe('general setup', () => {
      beforeAll(async () => {
        appTree = await generateDefaultWorkspace(schematicRunner, appTree);
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          {
            ...segmentRefsFeatureOptions,
            features: [TRACKING_PERSONALIZATION_FEATURE_NAME],
          },
          appTree
        );
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          segmentRefsFeatureOptions,
          appTree
        );
      });

      it('should install necessary Spartacus libraries', () => {
        const packageJson = JSON.parse(appTree.readContent('package.json'));
        let dependencies: Record<string, string> = {};
        dependencies = { ...packageJson.dependencies };
        dependencies = { ...dependencies, ...packageJson.devDependencies };

        for (const toAdd in peerDependencies) {
          if (
            !peerDependencies.hasOwnProperty(toAdd) ||
            toAdd === SPARTACUS_SCHEMATICS
          ) {
            continue;
          }
          const expectedDependency = dependencies[toAdd];
          expect(expectedDependency).toBeTruthy();
        }
      });

      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(segmentRefsFeatureModulePath);
        expect(module).toMatchSnapshot();
      });
    });

    describe('eager loading', () => {
      beforeAll(async () => {
        appTree = await generateDefaultWorkspace(schematicRunner, appTree);
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          {
            ...segmentRefsFeatureOptions,
            features: [TRACKING_PERSONALIZATION_FEATURE_NAME],
          },
          appTree
        );
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          { ...segmentRefsFeatureOptions, lazy: false },
          appTree
        );
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(segmentRefsFeatureModulePath);
        expect(module).toMatchSnapshot();
        expect(
          appTree.readContent(trackingPersonalizationFeatureModulePath)
        ).toBeTruthy();
      });
    });
  });
});
