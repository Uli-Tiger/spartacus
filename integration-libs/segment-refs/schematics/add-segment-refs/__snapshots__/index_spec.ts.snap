// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus segment-refs schematics: ng-add Segment-refs feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { SegmentRefsRootModule } from "@spartacus/segment-refs/root";

@NgModule({
  declarations: [],
  imports: [
    SegmentRefsRootModule
  ]
})
export class SegmentRefsFeatureModule { }
"
`;

exports[`Spartacus segment-refs schematics: ng-add Segment-refs feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { SegmentRefsRootModule } from "@spartacus/segment-refs/root";

@NgModule({
  declarations: [],
  imports: [
    SegmentRefsRootModule
  ]
})
export class SegmentRefsFeatureModule { }
"
`;
